bot/cogs/moderation/modlog.py

```diff
@@ -462,74 +476,27 @@ async def on_member_update(self, before: discord.Member, after: discord.Member)
             self._ignored[Event.member_update].remove(before.id)
             return
 
-        diff = DeepDiff(before, after)
-        changes = []
-        done = []
-
-        diff_values = {}
-
-        diff_values.update(diff.get("values_changed", {}))
-        diff_values.update(diff.get("type_changes", {}))
-        diff_values.update(diff.get("iterable_item_removed", {}))
-        diff_values.update(diff.get("iterable_item_added", {}))
+        changes = self.get_role_diff(before.roles, after.roles)
 
-        diff_user = DeepDiff(before._user, after._user)
-
-        diff_values.update(diff_user.get("values_changed", {}))
-        diff_values.update(diff_user.get("type_changes", {}))
-        diff_values.update(diff_user.get("iterable_item_removed", {}))
-        diff_values.update(diff_user.get("iterable_item_added", {}))
-
-        for key, value in diff_values.items():
-            if not key:  # Not sure why, but it happens
-                continue
-
-            key = key[5:]  # Remove "root." prefix
-
-            if "[" in key:
-                key = key.split("[", 1)[0]
+        # The regex is a simple way to exclude all sequence and mapping types.
+        diff = DeepDiff(before, after, exclude_regex_paths=r".*\[.*")
```

Comment:
I was going to suggest ignoring ABCs here, but it is is only for wherever deepdiff adds the brackets into the paths (didn't find that much in docs on the cases where it uses those). 
I'm not sure if there is a fitting abc and this seems to work nicely for the intended purposes, but the comment was a bit misleading since we want strings here which are also sequences.